src/mflux/server/openai_server.py

```diff
@@ -0,0 +1,243 @@
+from fastapi import FastAPI, HTTPException, Body
+from pydantic import BaseModel
+import base64
+import os
+import time
+import io
+from typing import List, Optional
+import logging
+import http.client
+import argparse
+import uvicorn
+
+from mflux.ui.defaults import WIDTH, HEIGHT
+from mflux import Config, StopImageGenerationException
+from mflux.callbacks.callback_registry import CallbackRegistry
+from mflux.callbacks.instances.memory_saver import MemorySaver
+from mflux.callbacks.instances.stepwise_handler import StepwiseHandler
+from common import ModelManager, create_temp_directory, cleanup_temp_file, MODEL_KEEP_ALIVE_FOREVER, MODEL_UNLOAD_IMMEDIATELY
+
+# Configure the logger
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+console_handler = logging.StreamHandler()
+console_handler.setFormatter(logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s'))
+logger.addHandler(console_handler)
+
+app = FastAPI()
+
+# Initialize the model manager
+model_manager = ModelManager()
+
+class GenerationRequest(BaseModel):
+    prompt: str
+    model: Optional[str] = "schnell"
+    base_model: Optional[str] = None
+    height: Optional[int] = HEIGHT
+    width: Optional[int] = WIDTH
+    size: Optional[str] = None
+    quality: Optional[str] = None
+    guidance: Optional[float] = 4.0
+    image_strength: Optional[float] = None
+    controlnet_strength: Optional[float] = None
+    seed: Optional[int] = None
+    quantize: Optional[int] = 8
+    n: Optional[int] = 1
+    lora_repo_id: Optional[str] = None
+    lora_scale: Optional[float] = 0.7
+    low_ram: Optional[bool] = False
+    stepwise_output: Optional[bool] = False
+    metadata: Optional[bool] = False
+    local_path: Optional[str] = None
+    response_format: Optional[str] = None
+    keep_alive: Optional[int] = MODEL_UNLOAD_IMMEDIATELY
```

Comment:
I don't think most of these should have `Optional` in them – that means Pydantic would also consider `None` a valid value for them. It will read the defaults if the user doesn't pass them in explicitly. 


```suggestion
    model: str = "schnell"
    base_model: Optional[str] = None
    height: Optional[int] = HEIGHT
    width: Optional[int] = WIDTH
    size: Optional[str] = None
    quality: Optional[str] = None
    guidance: float = 4.0
    image_strength: Optional[float] = None
    controlnet_strength: Optional[float] = None
    seed: Optional[int] = None
    quantize: int = 8
    n: int = 1
    lora_repo_id: Optional[str] = None
    lora_scale: float = 0.7
    low_ram: bool = False
    stepwise_output: bool = False
    metadata: bool = False
    local_path: Optional[str] = None
    response_format: Optional[str] = None
    keep_alive: int = MODEL_UNLOAD_IMMEDIATELY
```